// @bedrock-foundation/sdk/lib/cjs/models/shared.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenInfo = exports.TokenTypes = exports.createNonceSocketTopic = exports.TransactionStatuses = void 0;
var TransactionStatuses;
(function (TransactionStatuses) {
    TransactionStatuses["Unknown"] = "Unknown";
    TransactionStatuses["Pending"] = "Pending";
    TransactionStatuses["Scanned"] = "Scanned";
    TransactionStatuses["Confirmed"] = "Confirmed";
    TransactionStatuses["Error"] = "Error";
})(TransactionStatuses = exports.TransactionStatuses || (exports.TransactionStatuses = {}));
const createNonceSocketTopic = (nonce) => {
    return `nonce/${nonce}`;
};
exports.createNonceSocketTopic = createNonceSocketTopic;
var TokenTypes;
(function (TokenTypes) {
    TokenTypes["SOL"] = "SOL";
    TokenTypes["USDC"] = "USDC";
})(TokenTypes = exports.TokenTypes || (exports.TokenTypes = {}));
exports.TokenInfo = {
    [TokenTypes.SOL]: {
        sign: '◎',
        chainId: 103,
        address: 'So11111111111111111111111111111111111111112',
        symbol: 'SOL',
        name: 'Wrapped SOL',
        decimals: 9,
        logoURI: 'https://raw.githubusercontent.com/solana-labs/token-list/main/assets/mainnet/So11111111111111111111111111111111111111112/logo.png',
        extensions: {
            coingeckoId: 'solana',
        },
    },
    [TokenTypes.USDC]: {
        sign: '$',
        chainId: 101,
        address: 'EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v',
        symbol: 'USDC',
        name: 'USD Coin',
        decimals: 6,
        logoURI: 'https://raw.githubusercontent.com/solana-labs/token-list/main/assets/mainnet/EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v/logo.png',
        tags: [
            'stablecoin',
        ],
        extensions: {
            coingeckoId: 'usd-coin',
            serumV3Usdt: '77quYg4MGneUdjgXCunt9GgM1usmrxKY31twEy3WHwcS',
            website: 'https://www.centre.io/',
        },
    },
};
