// @bedrock-foundation/sdk/lib/cjs/actions/AuthorizationAction.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorizationAction = exports.authorizationDeliverySchema = exports.authorizationParamsSchema = void 0;
const joi_1 = __importDefault(require("joi"));
const web3_js_1 = require("@solana/web3.js");
const pay_1 = require("@solana/pay");
const JSURL = __importStar(require("@bedrock-foundation/jsurl"));
const JoiUtil = __importStar(require("../utils/JoiUtil"));
exports.authorizationParamsSchema = joi_1.default.object().keys({});
exports.authorizationDeliverySchema = joi_1.default.object().keys({
    account: joi_1.default.string().required(),
    params: joi_1.default.any(),
}).prefs({
    abortEarly: false,
});
class AuthorizationAction {
    constructor(basePath) {
        this.path = '/authorize';
        this.noncePath = '/authorize/nonce';
        this.basePath = basePath || 'https://localhost:3001';
    }
    createLink(params) {
        var _a;
        const requestRef = web3_js_1.Keypair.generate().publicKey.toBase58();
        params.refs = [requestRef, ...((_a = params === null || params === void 0 ? void 0 : params.refs) !== null && _a !== void 0 ? _a : [])];
        const url = `${this.basePath}${this.path}?params=${JSURL.stringify(params)}`;
        const nonceUrl = `${this.basePath}${this.noncePath}?params=${JSURL.stringify(params)}`;
        const link = (0, pay_1.encodeURL)({ link: new URL(url) }).toString();
        return {
            link,
            nonceUrl,
            refs: {
                requestRef,
            },
        };
    }
    createNonceLink(params) {
        return __awaiter(this, void 0, void 0, function* () {
            const result = this.createLink(params);
            if (!result.nonceUrl)
                return result;
            const response = yield fetch(result === null || result === void 0 ? void 0 : result.nonceUrl);
            const { nonce } = yield response.json();
            result.nonce = nonce;
            return result;
        });
    }
    validate(params) {
        return JoiUtil.validate(exports.authorizationParamsSchema, params);
    }
    validateDelivery(params) {
        return JoiUtil.validate(exports.authorizationDeliverySchema, params);
    }
}
exports.AuthorizationAction = AuthorizationAction;
