// @bedrock-foundation/sdk/lib/cjs/actions/EmptyWalletAction.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmptyWalletAction = exports.emptyWalletDeliverySchema = exports.emptyWalletParamsSchema = void 0;
const joi_1 = __importDefault(require("joi"));
const web3_js_1 = require("@solana/web3.js");
const pay_1 = require("@solana/pay");
const JoiUtil = __importStar(require("../utils/JoiUtil"));
exports.emptyWalletParamsSchema = joi_1.default.object().keys({});
exports.emptyWalletDeliverySchema = joi_1.default.object().keys({
    account: joi_1.default.string().required(),
    params: exports.emptyWalletParamsSchema,
}).prefs({
    abortEarly: false,
});
class EmptyWalletAction {
    constructor(basePath) {
        this.path = '/empty-wallet';
        this.basePath = basePath || 'https://localhost:3001';
    }
    createLink(_params) {
        const requestRef = web3_js_1.Keypair.generate().publicKey.toBase58();
        const url = `${this.basePath}${this.path}`;
        const urlWithRefs = `${url}&requestRef=${requestRef}`;
        const link = (0, pay_1.encodeURL)({ link: new URL(urlWithRefs) }).toString();
        return {
            link,
            refs: {
                requestRef,
            },
        };
    }
    validate(params) {
        return JoiUtil.validate(exports.emptyWalletParamsSchema, params);
    }
    validateDelivery(params) {
        return JoiUtil.validate(exports.emptyWalletDeliverySchema, params);
    }
}
exports.EmptyWalletAction = EmptyWalletAction;
