// @bedrock-foundation/sdk/lib/cjs/actions/TransferAction.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferAction = exports.transferDeliverySchema = exports.transferParamsSchema = void 0;
const joi_1 = __importDefault(require("joi"));
const web3_js_1 = require("@solana/web3.js");
const pay_1 = require("@solana/pay");
const JSURL = __importStar(require("@bedrock-foundation/jsurl"));
const JoiUtil = __importStar(require("../utils/JoiUtil"));
exports.transferParamsSchema = joi_1.default.object().keys({
    wallet: joi_1.default.string().required(),
    token: joi_1.default.string().required(),
    quantity: joi_1.default.number().optional(),
    size: joi_1.default.number().optional(),
    icon: joi_1.default.string().optional(),
    label: joi_1.default.string().optional(),
    refs: joi_1.default.array().items(joi_1.default.string()).default([]),
});
exports.transferDeliverySchema = joi_1.default.object().keys({
    account: joi_1.default.string().required(),
    params: exports.transferParamsSchema,
}).prefs({
    abortEarly: false,
});
class TransferAction {
    constructor(basePath) {
        this.path = '/transfer';
        this.basePath = basePath || 'https://localhost:3001';
    }
    createLink(params) {
        var _a;
        const requestRef = web3_js_1.Keypair.generate().publicKey.toBase58();
        params.refs = [requestRef, ...((_a = params === null || params === void 0 ? void 0 : params.refs) !== null && _a !== void 0 ? _a : [])];
        const url = `${this.basePath}${this.path}?params=${JSURL.stringify(params)}`;
        const link = (0, pay_1.encodeURL)({ link: new URL(url) }).toString();
        return {
            link,
            refs: {
                requestRef,
            },
        };
    }
    validate(params) {
        return JoiUtil.validate(exports.transferParamsSchema, params);
    }
    validateDelivery(params) {
        return JoiUtil.validate(exports.transferDeliverySchema, params);
    }
}
exports.TransferAction = TransferAction;
