// @bedrock-foundation/sdk/lib/esm/status/PollReferenceStatus.mjs
/* eslint-disable no-async-promise-executor */
import Joi from 'joi';
import * as JoiUtil from './../utils/JoiUtil.mjs';
export const PollReferenceQueryStringParamsSchema = Joi.object().keys({
    ref: Joi.string().required(),
});
export const pollReferenceStatusParamsSchema = Joi.object().keys({
    ref: Joi.string().required(),
    interval: Joi.number().optional(),
    maxRetries: Joi.number().optional(),
});
export class PollReferenceStatus {
    constructor(basePath) {
        this.path = '/poll-reference-status';
        this.basePath = basePath || 'https://localhost:3001';
    }
    async status(params) {
        const { ref } = params;
        const url = `${this.basePath}${this.path}?ref=${ref}`;
        const response = await fetch(url);
        return await response.json();
    }
    validate(params) {
        return JoiUtil.validate(pollReferenceStatusParamsSchema, params);
    }
    validateQueryStringParams(params) {
        return JoiUtil.validate(PollReferenceQueryStringParamsSchema, params);
    }
}
