// @bedrock-foundation/sdk/lib/esm/modules/BedrockCore.mjs
import { createLink, createNonceLink, } from './../models/createLink.mjs';
export class BedrockCore {
    constructor(params) {
        var _a;
        this.createTransferLink = (params) => {
            return createLink(this.basePath, BedrockCore.Paths.Transfer, params);
        };
        this.createEmptyWalletLink = (params) => {
            return createLink(this.basePath, BedrockCore.Paths.EmptyWallet, params);
        };
        this.createAuthorizationNonceLink = async (params) => {
            return await createNonceLink(this.basePath, BedrockCore.Paths.Authorization, BedrockCore.Paths.AuthorizationNonce, params);
        };
        this.getReferenceStatus = async (params) => {
            const { ref } = params;
            const url = `${this.basePath}${BedrockCore.Paths.ReferenceStatus}?ref=${ref}`;
            const response = await fetch(url);
            return await response.json();
        };
        this.basePath = (_a = params.basePath) !== null && _a !== void 0 ? _a : 'https://pay.bedrock.fyi';
    }
}
BedrockCore.Paths = {
    Transfer: '/transfer',
    EmptyWallet: '/empty-wallet',
    Authorization: '/authorization',
    AuthorizationNonce: '/authorization/nonce',
    ReferenceStatus: '/reference-status',
};
