// @bedrock-foundation/sdk/lib/esm/actions/EmptyWalletAction.mjs
import Joi from 'joi';
import { Keypair } from '@solana/web3.js';
import { encodeURL } from '@solana/pay';
import * as JoiUtil from './../utils/JoiUtil.mjs';
export const emptyWalletParamsSchema = Joi.object().keys({});
export const emptyWalletDeliverySchema = Joi.object().keys({
    account: Joi.string().required(),
    params: emptyWalletParamsSchema,
}).prefs({
    abortEarly: false,
});
export class EmptyWalletAction {
    constructor(basePath) {
        this.path = '/empty-wallet';
        this.basePath = basePath || 'https://localhost:3001';
    }
    createLink(_params) {
        const requestRef = Keypair.generate().publicKey.toBase58();
        const url = `${this.basePath}${this.path}`;
        const urlWithRefs = `${url}&requestRef=${requestRef}`;
        const link = encodeURL({ link: new URL(urlWithRefs) }).toString();
        return {
            link,
            refs: {
                requestRef,
            },
        };
    }
    validate(params) {
        return JoiUtil.validate(emptyWalletParamsSchema, params);
    }
    validateDelivery(params) {
        return JoiUtil.validate(emptyWalletDeliverySchema, params);
    }
}
